player.py

```diff
@@ -15,6 +15,7 @@ def __init__(self, name, shipPlacementStrategy: IShipPlacementStrategy) -> None:
         self.name = name
         self._score = 0
         self._hits = 0
+        self.num_ships = 4
```

Comment:
Since the game is designed to have any number of ships. It would be a good idea if we set num_ships through the Player constructor.